demo/advanced_demo3.py

```diff
@@ -46,6 +47,7 @@
 import random
 import time
 import socket
+from six.moves import range as xrange
```

Comment:
I wonder if we really need xrange at all. In py2, `range` exists and works the same as py3's `range` except deep inside py3's `range` returns an iterator. But does it matter? I doubt we really needed it to be `xrange` in py2.